accord-core/src/main/java/accord/local/Bootstrap.java

```diff
@@ -261,7 +261,11 @@ private void safeToReadCallback(SafeToRead state, Timestamp executeAt, Throwable
             else
             {
                 // TODO (expected): first check to see if we are still relevant
-                node.agent().onFailedBootstrap("SafeToRead", state.ranges, () -> started(state, null), failure);
+                CommandStore store = CommandStore.current();
+                node.agent().onFailedBootstrap("SafeToRead", state.ranges, () -> {
+                    if (CommandStore.maybeCurrent() == store) started(state, null);
```

Comment:
maybe we should have a method on `CommandStore` that accepts a `self` parameter and either executes the `Runnable` or `submits` it as necessary